entity/c/c_api.c

```diff
@@ -0,0 +1,197 @@
+#include "c_api.h"
+
+extern int sent_seq_num;
+extern unsigned char entity_client_state;
+extern unsigned char entity_server_state;
+extern long int st_time;
+
+config_t *load_config(char *path)
+{
+    return load_config_t(path);
+}
+
+session_key_t *get_session_key(config_t *config_info)
+{
+    if (strncmp((const char *)config_info->network_protocol, "TCP", 3) == 0)
+    {
+        return send_session_key_req_via_TCP(config_info);
+    }
+    else if (strncmp((const char *)config_info->network_protocol, "UDP", 3) == 0)
+    {
+        return send_session_key_req_via_UDP();
+    }
+    return 0;
+}
+
+int secure_connect_to_server(session_key_t *s_key, config_t *config_info)
+{
+    int sock;
+    connect_as_client((const char *)config_info->entity_server_ip_addr, (const char *)config_info->entity_server_port_num, &sock);
+    unsigned char entity_nonce[HS_NONCE_SIZE];
+    unsigned int parsed_buf_length;
+    unsigned char *parsed_buf = parse_handshake_1(s_key, entity_nonce, &parsed_buf_length);
+    unsigned char sender[128]; // TODO: actually only needs 19 bytes.
+    unsigned int sender_length;
+    make_sender_buf(parsed_buf, parsed_buf_length, SKEY_HANDSHAKE_1, sender, &sender_length);
+    write(sock, sender, sender_length);
+    free(parsed_buf);
+    entity_client_state = HANDSHAKE_1_SENT;
+
+    // received handshake 2
+    unsigned char received_buf[1000];
+    unsigned int received_buf_length = read(sock, received_buf, sizeof(received_buf));
+    unsigned char message_type;
+    unsigned int data_buf_length;
+    unsigned char *data_buf = parse_received_message(received_buf, received_buf_length, &message_type, &data_buf_length);
+    if (message_type == SKEY_HANDSHAKE_2)
+    {
+        if (entity_client_state != HANDSHAKE_1_SENT)
+        {
+            printf("Comm init failed: wrong sequence of handshake, disconnecting...\n");
+        }
+        unsigned int parsed_buf_length;
+        unsigned char *parsed_buf = check_handshake_2_send_handshake_3(data_buf, data_buf_length, entity_nonce, s_key, &parsed_buf_length);
+        unsigned char sender[256];
+        unsigned int sender_length;
```

Comment:
`sender` and `sender_length` are re-defined again (already defined at line 33, 34).
I know the `sender` (client's name) here and the `sender` (server's name) at line 33 are different. Then, it will make sense to have different variable names to avoid implicitly overriding the variables.